### ISSUE TITLE
Cannot get eclox doxygen to run in the code composer studio eclipse package

### ISSUE THREAD

**Issue by xtracold** [anb0s/eclox#220]
**Headline**

Cannot get eclox doxygen to work in the code composer studio package of eclipse. 
**Description**
In code composer studio v7.1.x I installed eclox from the Eclipse marketplace but under preferences the doxygen version is not filled in and generating a doxyfile always fails. Checking the logs it seems that doxygen cannot be found/executed. 

Would appreciate any guidance on finding the root cause.

[log.txt](https://github.com/anb0s/eclox/files/1503629/log.txt)

- Plugin version: 1.8.13
- Eclipse Version: 4.6.0 (Code composer studio 7.1.x
- Java Version: 1.8.0_112
- Operating System Version: Windows 10

**Comment by anb0s** [comment]
The Eclox Doxygen (bundle) version is 1.8.13, what is the Eclox version itself? v0.11.2 or older?
I see that doxygen executable cannot be found, but if you have the bundled version installed it should work:
> java.io.IOException: Cannot run program "doxygen": CreateProcess error=2, The system cannot find the file specified
java.lang.RuntimeException: Unable to get doxygen version: 

I will investigate...

**Comment by anb0s** [comment]
Plese check if under "Preferences->Doxygen" use see the Bundled version selected:
![image](https://user-images.githubusercontent.com/95811/33239053-4fb5c89a-d29a-11e7-8318-494e51a2a9a9.png)

**Comment by xtracold** [comment]
I've upgraded now to CCS 7.2.x and reinstalled eclox from the marketplace with the same issue. I also noticed that the logs were showing doxygen was not possible to execute.
Here are the installed versions:
![image](https://user-images.githubusercontent.com/33994750/33242449-16f11164-d28a-11e7-9c8e-ed006455c421.png)

Because it cannot find or execute the executable it also is unable to list the available versions under preferences:

![image](https://user-images.githubusercontent.com/33994750/33242460-3da787fc-d28a-11e7-8f2e-7b039b457e99.png)

Maybe this is as simple as a permissions issue on the install directories due to me using Ti's code composer 7 version of Eclipse?

**Comment by xtracold** [comment]
I think the issue was with the install of the bundled doxygen. I've installed manually from http://www.stack.nl/~dimitri/doxygen/download.html and added the executable manually under preferences. I'm now able to generate the docs. 

If you want me to check anything in my config to help understand why the bundled install was not accessible let me know.

![image](https://user-images.githubusercontent.com/33994750/33242682-9cad6a20-d28d-11e7-883c-96334a6f3c14.png)

**Comment by anb0s** [comment]
Thanks! May be your eclipse (Code composer studio) or workspace is located in "C:\Program Files" and eclipse cannot extract the bundle or cannot execute the doxygen.exe in this location. Please start ypur eclipse as Administrator and provide me the new logs.

I will copy my eclipse and workspace to such folders and try to reproduce...

**Comment by xtracold** [comment]
My CCS workspace is under my profile here:
C:\Users\jwardlaw\workspace_v7
The install of CCS is to C:\program files
Under preferences the default doxygen version now shows, you will note it did not previously even after I installed doxygen direct from the doxygen site. This could be due to a restart though. The bundled version is still not picked up. 

![image](https://user-images.githubusercontent.com/33994750/33279510-a54a9394-d353-11e7-947f-24a3e2d7c285.png)

Here the log file from an administrator startup.

[.log](https://github.com/anb0s/eclox/files/1506846/default.log)

Looks like eclox still couldn't access the bundled version? All very strange, at least I can run with the unbundled version.

**Comment by anb0s** [comment]
Still no idea whats wrong...

**Comment by anb0s** [comment]
I've now a Widows 7 PC where it's reproducible with just release v0.120. I will investigate more...

![image](https://user-images.githubusercontent.com/95811/33932059-34da3780-dff2-11e7-813d-0aa1829481d7.png)

```
org.gna.eclox.core
Error
Wed Dec 13 10:40:19 CET 2017
Exception caught. java.lang.RuntimeException: Unable to get doxygen version: 

java.lang.RuntimeException: Unable to get doxygen version: 
	at eclox.core.doxygen.Doxygen.getVersion(Doxygen.java:165)
	at eclox.core.ui.DefaultDoxygenFieldEditor.updateItem(DefaultDoxygenFieldEditor.java:156)
	at eclox.core.ui.DefaultDoxygenFieldEditor.addItem(DefaultDoxygenFieldEditor.java:132)
	at eclox.core.ui.DefaultDoxygenFieldEditor.doLoad(DefaultDoxygenFieldEditor.java:380)
	at org.eclipse.jface.preference.FieldEditor.load(FieldEditor.java:495)
	at org.eclipse.jface.preference.FieldEditorPreferencePage.initialize(FieldEditorPreferencePage.java:306)
	at org.eclipse.jface.preference.FieldEditorPreferencePage.createContents(FieldEditorPreferencePage.java:231)
	at org.eclipse.jface.preference.PreferencePage.createControl(PreferencePage.java:241)
	at org.eclipse.jface.preference.PreferenceDialog.createPageControl(PreferenceDialog.java:1430)
	at org.eclipse.jface.preference.PreferenceDialog$9.run(PreferenceDialog.java:1197)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.ui.internal.JFaceUtil$1.run(JFaceUtil.java:50)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:173)
	at org.eclipse.jface.preference.PreferenceDialog.showPage(PreferenceDialog.java:1189)
	at org.eclipse.ui.internal.dialogs.FilteredPreferenceDialog.showPage(FilteredPreferenceDialog.java:608)
	at org.eclipse.jface.preference.PreferenceDialog$5.lambda$0(PreferenceDialog.java:656)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
	at org.eclipse.jface.preference.PreferenceDialog$5.selectionChanged(PreferenceDialog.java:653)
	at org.eclipse.jface.viewers.StructuredViewer$3.run(StructuredViewer.java:878)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.ui.internal.JFaceUtil$1.run(JFaceUtil.java:50)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:173)
	at org.eclipse.jface.viewers.StructuredViewer.firePostSelectionChanged(StructuredViewer.java:875)
	at org.eclipse.jface.viewers.StructuredViewer.setSelection(StructuredViewer.java:1729)
	at org.eclipse.jface.viewers.TreeViewer.setSelection(TreeViewer.java:1077)
	at org.eclipse.jface.preference.PreferenceDialog.selectSavedItem(PreferenceDialog.java:991)
	at org.eclipse.ui.internal.dialogs.FilteredPreferenceDialog.selectSavedItem(FilteredPreferenceDialog.java:656)
	at org.eclipse.jface.preference.PreferenceDialog.lambda$0(PreferenceDialog.java:346)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
	at org.eclipse.jface.preference.PreferenceDialog.createContents(PreferenceDialog.java:343)
	at org.eclipse.jface.window.Window.create(Window.java:426)
	at org.eclipse.jface.dialogs.Dialog.create(Dialog.java:1095)
	at org.eclipse.ui.internal.dialogs.WorkbenchPreferenceDialog.createDialogOn(WorkbenchPreferenceDialog.java:102)
	at org.eclipse.ui.dialogs.PreferencesUtil.createPreferenceDialogOn(PreferencesUtil.java:192)
	at org.eclipse.ui.dialogs.PreferencesUtil.createPreferenceDialogOn(PreferencesUtil.java:116)
	at org.eclipse.ui.internal.OpenPreferencesAction.run(OpenPreferencesAction.java:62)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:473)
	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:565)
	at org.eclipse.jface.action.ActionContributionItem.lambda$4(ActionContributionItem.java:397)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4418)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1079)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4236)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3824)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1121)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1022)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:150)
	at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:693)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:610)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:138)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1519)


```

**Comment by anb0s** [comment]
Verified that it's working @ Linux (Debian 10)
![image](https://user-images.githubusercontent.com/95811/34302805-cc9822ac-e732-11e7-8442-7e52369536e7.png)

--> Windows only issue...

**Comment by aslepp** [comment]
Hello!
I have the same problem.
I'm working with Windows 7 64-bit.
I think, the problem is the absent libclang.dll.
If you extract all that eclox-bundle have, and try to launch doxygen.exe, you could see a System Error about it.
In the original Doxygen zip is libclang.dll included, and it works.

**Comment by anb0s** [comment]
Hello! Thank You @aslepp !
I think it's explainable now:
![image](https://user-images.githubusercontent.com/95811/35038515-532b92c0-fb7b-11e7-99b2-407d4d81b33d.png)

I will provide new bundle plugin asap..

**Comment by anb0s** [comment]
it will be fixed with #222 and #223

**Comment by anb0s** [comment]
released with eclox 0.12.1.v20180118-1753 and eclox doxygen 1.8.13.v20180118-1753
